overlord/snapstate/check_snap.go

```diff
@@ -181,6 +181,11 @@ func checkCoreName(st *state.State, snapInfo, curInfo *snap.Info, flags Flags) e
 		return err
 	}
 
+	// FIMXE: make this more elegant
+	if core.Name() == "ubuntu-core" {
```

Comment:
I'm not sure I understand this one,  I thought you would need the other direction (allowing core while ubuntu-core is there?) also this as it is will let people install ubuntu-core again, which we might not want